In [1]:
import torch
print("CUDA Available:", torch.cuda.is_available())  # Should print True
print("GPU:", torch.cuda.get_device_name(0))  # Should print your GPU name
print("CUDA Version:", torch.version.cuda)  # Should match CUDA 12.1
print("PyTorch Version:", torch.__version__)  # Should match installed version


CUDA Available: True
GPU: NVIDIA GeForce RTX 4080 SUPER
CUDA Version: 12.1
PyTorch Version: 2.5.1


In [20]:
import torch
import time

# Move computations to GPU
x = torch.randn(10000, 10000, device="cuda")
y = torch.randn(10000, 10000, device="cuda")

# Measure execution time
start = time.time()
z = torch.matmul(x, y)
torch.cuda.synchronize()  # Ensures all computations finish before measuring time
end = time.time()

print(f"Matrix multiplication completed in {end - start:.4f} sec on GPU")


Matrix multiplication completed in 0.0732 sec on GPU


PC, win 11, Specs:
- CUDA Available: True
- GPU: NVIDIA GeForce RTX 4080 SUPER
- CUDA Version: 12.1
- PyTorch Version: 2.5.1

Matrix multiplication completed in 0.0937 sec on GPU